src/workerd/api/queue.c++

```diff
@@ -147,7 +147,7 @@ QueueMessage::QueueMessage(
     v8::Isolate* isolate, IncomingQueueMessage message, IoPtr<QueueEventResult> result)
     : id(kj::mv(message.id)),
       timestamp(message.timestamp),
-      body(isolate, jsg::Deserializer(isolate, kj::mv(message.body)).readValue()),
+      body(isolate, jsg::Deserializer(isolate, message.body.asPtr()).readValue()),
```

Comment:
Does this actually fix the problem for you? The only way I can figure that this would change anything is if the `auto` type inference on `jsg::Deserializer` is choosing to treat `data` as a `kj::Array<kj::byte>` rather than as a `kj::Array<kj::byte>&&` or `kj::Array<kj::byte>&`...

Which I guess is possible, although surprising.

Although in either case, if this fixes it for you, I'm ok with merging it. It certainly shouldn't make anything worse.